app/src/main/java/com/conupods/OutdoorMaps/Services/OutdoorDirectionsService.java

```diff
@@ -0,0 +1,128 @@
+package com.conupods.OutdoorMaps.Services;
+
+import android.app.Activity;
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.conupods.R;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Polyline;
+import com.google.android.gms.maps.model.PolylineOptions;
+import com.google.maps.DirectionsApiRequest;
+import com.google.maps.GeoApiContext;
+import com.google.maps.PendingResult;
+import com.google.maps.internal.PolylineEncoding;
+import com.google.maps.model.DirectionsResult;
+import com.google.maps.model.DirectionsRoute;
+import com.google.maps.model.TravelMode;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+
+public class OutdoorDirectionsService {
+
+    private GoogleMap mMap;
+    private GeoApiContext gac;
+
+    private Activity passedActivity;
```

Comment:
this shouldn't be a member variable, you arent using it outside of the constructor

Reply:
removed passedActivity as a member variable